src/cluster_copier.cc

```diff
@@ -330,14 +330,21 @@ namespace grpc_labview {
             {
                 auto nestedMetadata = repeatedNested->_value.front()->_metadata;
                 auto clusterSize = nestedMetadata->clusterSize;
+                auto byteSize = repeatedNested->_value.size() * clusterSize;
+                auto alignment = nestedMetadata->alignmentRequirement;
+                auto alignedElementSize = byteSize / alignment;
+                if (byteSize % alignment != 0)
+                {
+                    alignedElementSize++;
+                }
 
-                NumericArrayResize(0x08, 1, start, repeatedNested->_value.size() * clusterSize);
+                NumericArrayResize(GetTypeCodeForSize(alignment), 1, start, alignedElementSize);
```

Comment:
Note: The previous allocation was 8x larger than it needed to be since the element size passed was the byte size, and we were specifying an element size of 8 bytes. We now pick an element size that meets our alignment requirements and update the number of elements to allocate accordingly.